pyiceberg/table/__init__.py

```diff
@@ -1371,6 +1375,45 @@ def to_polars(self) -> pl.LazyFrame:
 
         return pl.scan_iceberg(self)
 
+    def delete_orphaned_files(self) -> None:
```

Comment:
I think it would be good to add some options that we also have [on the Java side](https://iceberg.apache.org/docs/nightly/spark-procedures/#remove_orphan_files), at a minimum:

- `older_than`: Remove orphan files created before this timestamp (Defaults to 3 days). It can be that some process is writing to the table, and has some files staged to be added to the metadata tree. If we don't take this into account, it might be that these files are removed in the period between writing and committing.
- `dry_run`: When true, don't actually remove files (defaults to false). I think it would be nice to return a set of the number of files removed:
```suggestion
    def delete_orphaned_files(self) -> Set[str]:
```

Reply:
Added!

Reply:
Is there a reason that `older_than` is not a table property?

Reply:
Erm, I think I was messing up two PRs :D